### PR TITLE
Cleaning up some warnings from tests and React runtime

### PR DESCRIPTION
**Pull request by olemartinorg**
## Description
This silences some knows warnings in unit tests, and tests for them instead of outputting them to the console. In addition, it cleans up some warnings shown in the console when running some apps.

## Verification
- [x] **Your** code builds clean without any errors or warnings
- [x] Manual testing done (required)
- [x] Relevant automated test added (if you find this hard, leave it and we'll help out)
- [x] All tests run green

## Documentation
- ~~[ ] User documentation is updated with a separate linked PR in [altinn-studio-docs.](https://github.com/Altinn/altinn-studio-docs) (if applicable)~~

### CODE REVIEW

File: src/altinn-app-frontend/src/components/base/ParagraphComponent.tsx [hunk 1]
```diff
@@ -34,6 +34,11 @@ const useStyles = makeStyles({
 
 export function ParagraphComponent(props: IComponentProps) {
   const classes = useStyles();
+  const isHeader =
+    typeof props.text === 'object' &&
+    typeof (props.text as any).type === 'string' &&
+    (props.text as any).type.match(/^h\d+$/);
+
```
Comment (haakemon) [inline src/altinn-app-frontend/src/components/base/ParagraphComponent.tsx:41]: I think this deserves a unit test 🙂

**Review by haakemon — CHANGES_REQUESTED** [review]

File: src/altinn-app-frontend/src/components/base/ParagraphComponent.test.tsx [hunk 2]
```diff
@@ -58,4 +58,19 @@ describe('ParagraphComponent', () => {
...
+        id={mockId}
+        text={<h3>Hello world</h3>}
+        language={mockLanguage}
+        getTextResource={mockGetTextResource}
+        textResourceBindings={mockTextResourceBindings}
+        {...({} as IComponentProps)}
+      />,
+    );
+
+    expect(shallowParagraphComponent).toMatchSnapshot();
```
[the remark this answers is not in the document]
Comment (olemartinorg) [reply src/altinn-app-frontend/src/components/base/ParagraphComponent.test.tsx:74]: You shouldn't be sorry! I'm a complete newbie when it comes to snapshot testing, so I hadn't considered problems like these. I fully agree.

Yep, there's potentially lots of things that shouldn't be inside a `<p>`, but I think headers were the most common case here. Either this evolves into a huge complexity, or we should just use `<div>` by default when the text is a react element. What do you think?

[the remark this answers is not in the document]
Comment (haakemon) [reply src/altinn-app-frontend/src/components/base/ParagraphComponent.test.tsx]: Thanks for updating it, looks much better now 😁
I think fixing the immediate problem (checking for headers) is enough for now. There are so many cases to handle that I'm not really sure the approach where Paragraph component accepts any content as children is the best. It should probably only accept a string. But changing that would be breaking, so not something we can do easily - but at least something we need to consider for the next major version.

**Review by haakemon — APPROVED** [review]
🥇

**Pull request merged**